completion_aggregator/views.py

```diff
@@ -30,9 +32,11 @@ def get(self, request, course_key, chapter_id=None):
         if chapter_id is not None:
             new_req['requested_fields'] = "chapter"
         request.GET = new_req
-        completion_resp = CompletionDetailView.as_view()(request, course_key).data
-        if completion_resp:
-            results = completion_resp.get('results')
+        with transaction.atomic():
+            completion_resp = CompletionDetailView.as_view()(request, course_key)
+
+        if completion_resp.response_code == status.HTTP_200_OK:
```

Comment:
@keithgg The previous step is not actually an HTTP request, so it doesn't contain a "response_code" attribute. Testing this out on production resulted in the following error:
```
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 554, in wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 52, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 944, in wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/views/decorators/clickjacking.py", line 50, in wrapped_view
    resp = view_func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/completion_aggregator/views.py", line 38, in get
    if completion_resp.response_code == status.HTTP_200_OK:
AttributeError: 'Response' object has no attribute 'response_code'
```

Can you revert it back to the old check? Just `if completion_resp:` should be good enough.

Reply:
Sure. PR updated.